### PR TITLE
My version and updates. Check the QuestionEndpoint and QuestionFilter…

### PR DESCRIPTION
**Pull request by dkheyman**
…Endpoint only

### CODE REVIEW

File: service.py [hunk 1]
```diff
@@ -45,6 +45,14 @@
             static_url_path='/flask-apispec/static'
 )
 
+#app.config.from_pyfile('db-config.cfg')
```
Comment (wfrece) [inline service.py]: Fine to bring this back but I'm going to comment it out in my next commit. We don't use this (yet...). I'll explain next time we chat.

File: view/core.py [hunk 2]
```diff
@@ -114,47 +114,51 @@ def delete(self, respondent_id):
...
-        question_kwargs = request.args.to_dict()
-
+        question_kwargs = request.json
         results = c.update_question(
             question_id=question_id,
             **question_kwargs
         )
         return make_response(jsonify(data=results), 201)
 
+    @marshal_with(QuestionSchema)
```
Comment (wfrece) [inline view/core.py]: Why marshal a delete? Not even the examples do this...

File: view/core.py [hunk 3]
```diff
@@ -114,47 +114,51 @@ def delete(self, respondent_id):
...
             question_id=question_id
         )
         return make_response(jsonify(data=results), 200)
 
+    #@use_kwargs(QuestionSchema, apply=True)
+    @marshal_with(QuestionSchema)
     def post(self, question_id):
-        question_kwargs = request.args.to_dict()
-
+        question_kwargs = request.json
```
Comment (wfrece) [inline view/core.py]: Now we are unwinding this? This fixed an issue a while back...now it breaks stuff?

File: view/core.py [hunk 4]
```diff
@@ -256,34 +260,48 @@ def delete(self, answer_id):
 def handle_validation_error_exception(e):
     '''Handler for the ValidationError exception thrown by MongoEngine'''
     status_code = 400
-    return {
+    return make_response(jsonify({
         'exception_type': type(e).__name__,
-        'status_code'   : status_code
-    }, status_code
+        'status_code'   : status_code,
+        'exception'     : str(e)
```
Comment (wfrece) [inline view/core.py]: surprised I didn't think of this before, good add.

File: view/core.py [hunk 5]
```diff
@@ -256,34 +260,48 @@ def delete(self, answer_id):
...
         'status_code'   : status_code
-    }, status_code
+    }), status_code)
 
 @current_app.errorhandler(ValueError)
 def handle_value_error_exception(e):
     '''Handler for the ValueError exception'''
     status_code = 400
-    return {
+    return make_response(jsonify({
```
Comment (wfrece) [inline view/core.py]: I dont know why this worked before, but this definitely looks right regardless.

File: view/core.py (hunk 2 above)
Comment (dkheyman) [reply view/core.py]: True, I guess i thought it would make sense. I can remove.

**Pull request merged**